CMakeLists.txt

```diff
@@ -1,26 +1,59 @@
+#Forces the build system to have cmake from a certain version above!
 cmake_minimum_required(VERSION 3.9)
 
+project(latte VERSION 1.2.2 LANGUAGES Fortran)
+
+#Enables the -DPROGRESS=<on/off> option?
 option(PROGRESS "Use progress library" OFF)
 
-project(latte VERSION 1.2.2 LANGUAGES Fortran)
+#Since progress depends on BML and bml is writtend in C, then:
 if(PROGRESS)
   enable_language(C CXX)
 endif()
 
+if(PROGRESS)
+  include(FindPkgConfig)
+  #Looks for the installation of progress in CMAKE_MODULE_PATH or CMAKE_PREFIX_PATH or whatever else cmake needs!
+  find_package(PROGRESS CONFIG QUIET)
+  #Checks if the PROGRESS modules are indeed available
+  pkg_check_modules(PROGRESS REQUIRED progress)
```

Comment:
This is duplicated, either uses `pkg_check_modules()` or `find_package(... CONFIG)`.